languages/common-lisp/exercises/concept/sameness/sameness.lisp

```diff
@@ -0,0 +1,13 @@
+(defpackage sameness
+  (:use :cl)
+  (:export :robot))
+
+(in-package :sameness)
+
+(defun robot (room-id)
+  "Return a key to use for the doors in the room designated by ROOM-ID."
+  (case room-id
+    ;; add forms such as:
+    ;; (room-id-13 #'equalp)
+
+    (t (constantly nil))))
```

Comment:
It was not my intention to teach the student `case` in this exercise - so this stub (with example that will not work for the exercise) was intended to give them the space to 'fill in the blanks' i.e. choosing the correct equality predicate for the room.

I guess I could add some brief instructions for `case` to the instructions file. And/or make sure this exercise has a prerequisite upon a concept that would teach them `case`.

Reply:
> I guess I could add some brief instructions for case to the instructions file. And/or make sure this exercise has a prerequisite upon a concept that would teach them case.

My preference would be for the latter. @TheLostLambda?

Reply:
Heya! I think this exercise should already depend on the `conditionals` exercise, which, while not yet implemented, should teach case. So the student _should_ know about `case` by the time they make it to this exercise.

I agree with @verdammelt that it's not really the focus of this exercise though, so I don't have issues with stubbing it out either. I'm ultimately impartial though, so I'd be fine keeping or ditching the stub.

Reply:
I can see the argument for keeping it in. My personal preference is to keep it out, but I have something against methods with stubs inside the methods. This is a remnant of some horrible courses I've taken where they used this "fill in the blanks" method, which I found awful to work with. But I'd be fine with keeping it in. Whichever you prefer.

Reply:
I think either:
1. Leave this in.
2. Split into into different methods rather than a case statement.

My rationale is that we want to somehow delineate the tasks. Atm, we're doing that through a case rather than different functions. Splitting into different functions is more "normal" for v3, but if that doesn't make sense for this exercise, then the case effectively becomes the collection of function stubs that we normally have.

I don't speak lisp so I don't know what's most sensible, but I think we need *some* scaffolding for each exercise and here that could be the case if not functions.

But I very much agree with @ErikSchierboom's instincts against having boilerplate "fill in the blanks" as a general rule.